tests/conftest.py

```diff
@@ -27,3 +30,37 @@ def pytest_configure(config):
     if _is_in_ci() and _is_pip_installed():
         print("***** Installed packages *****", flush=True)
         subprocess.check_call([sys.executable, '-m', 'pip', 'freeze', '--all'])
+    if config.pluginmanager.hasplugin("xdist"):
+        config.pluginmanager.register(DeferPlugin())
+
+
+# https://docs.pytest.org/en/latest/how-to/writing_hook_functions.html#optionally-using-hooks-from-3rd-party-plugins
+class DeferPlugin:
+    """Simple plugin to defer pytest-xdist hook functions."""
+
+    # Edit the environment variable `CUDA_VISIBLE_DEVICES` for each session.
+    # Cannot use `pytest_configure_node` nor `pytest_testnodeready` hook,
+    # because they are called in the `master` node (process).
+    # See also https://github.com/pytest-dev/pytest-xdist/issues/179.
+    @pytest.fixture(autouse=True, scope='session')
+    def _rotate_cuda_visible_devices(self, worker_id):
+        n_gpu = os.environ.get('CUPY_TEST_GPU_LIMIT')
```

Comment:
Can we ensure that CUDA is not loaded (so that we can overwrite `CUDA_VISIBLE_DEVICES`)? Maybe `assert 'cupy' not in sys.modules` can be used.

Reply:
I tried
```python
        assert not {
            'cupy', 'cupy_backends', 'cupyx',
        }.intersection(sys.modules)
```
in this function, but it failed.  However, `CUDA_VISIBLE_DEVICES` looks working to me because `cupy.cuda.Device(0).attributes["PciBusId"]` returned different values.

I didn't come up with an idea for testing this in a portable manner.

Reply:
I see, thanks for testing!